### ISSUE TITLE
Unable to get slices including the MSB

### ISSUE THREAD

**Issue by benbr8** [AlexAltea/htypes#6]
It is not possible to get a slice from a nint including the most significant bit:
```python
>>> from nativetypes import uint8
>>> u8=uint8(1<<7)
>>> bin(u8)
'0b10000000'
>>> u = u8[7:8]
Traceback (most recent call last):
  File "native_int.py", line 121, in __getitem__
    assert 0 <= key.stop < self.b
AssertionError
```
IMO the assertion in `__getitem__` is wrong and should be instead
```python
assert 0 <= key.stop <= self.b
```
since the last element index is actually `key.stop-1` in Python

This seems to work:
```python
>>> from nativetypes import uint8
>>> u8=uint8(1<<7)
>>> bin(u8)
'0b10000000'
>>> u = u8[7:8]
>>> u
nint(1)
```

**Comment by AlexAltea** [comment]
> IMO the assertion in `__getitem__` is wrong and should be instead
> ```
> assert 0 <= key.stop <= self.b
> ```

Nice catch, thank you! I've fixed this issue in 3268ff928979de61cb921afdd09499746b0f7f29 and bumped the PIP package to v1.0.4: https://pypi.org/project/nativetypes/1.0.4/

**Comment by benbr8** [comment]
Thanks for the fast fix!
